sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -490,10 +621,26 @@ trait DivModLike extends BinaryArithmetic {
       s"${eval2.value} == 0"
     }
     val javaType = CodeGenerator.javaType(dataType)
-    val operation = if (operandsDataType.isInstanceOf[DecimalType]) {
-      decimalToDataTypeCodeGen(s"${eval1.value}.$decimalMethod(${eval2.value})")
+    val checkOverflow = if (operandsDataType.isInstanceOf[DecimalType]) {
+      val decimal = super.dataType.asInstanceOf[DecimalType]
+      val errorContextCode = if (nullOnOverflow) {
+        "\"\""
+      } else {
+        ctx.addReferenceObj("errCtx", queryContext)
+      }
+      val decimalValue = ctx.freshName("decimalValue")
+      // scalastyle:off line.size.limit
+      s"""
+         |${CodeGenerator.javaType(decimal)} $decimalValue = ${eval1.value}.$decimalMethod(${eval2.value}).toPrecision(
```

Comment:
```suggestion
         |${ev.value} = ${eval1.value}.$decimalMethod(${eval2.value}).toPrecision(
```
We can assign a variable twice in java.

Reply:
The data type may different, e.g. the ev of `IntegralDivide`  is long but `Divide` is decimal if the input type is decimal. So here use a temp decimal type varible.